### ISSUE TITLE
Cross to clear input field does not work on result page

### ISSUE THREAD

**Issue by pro-sumer** [searxng/searxng#3725]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**
 2024.8.9+f1c05e7c1

**How did you install SearXNG?**
searxng-docker

**What happened?**
Cross to clear input field does not work on result page (does work on start page) 

**How To Reproduce**
1. Search for any text
2. Click on the cross in the input field on the results page

**Expected behavior**
Input field is cleared after clicking on cross

**Screenshots & Logs**
![Input@2x](https://github.com/user-attachments/assets/5f75db2c-a671-4e53-aeef-3241ff614ec6)

**Additional context**
Seems to work fine on public instances?

**Comment by return42** [comment]
For me it works .. can you please test it on https://darmarit.org/searx/search?q=foo .. I tested with FFox and Chromium .. does the issue appears only in Safari?

**Comment by pro-sumer** [comment]
Yes, it works fine on https://darmarit.org/searx/search?q=foo in Safari on Mac.

I experienced the issue with my instance both in Safari (Mac) and Mobile Safari (iPhone/iPad), but I just tested my instance in Firefox and Edge on a Windows PC. The cross does not work there either, so it's probably not the browser, but something with (the settings of?) my instance.

Hope to do some more experimenting this weekend.

**Comment by pro-sumer** [comment]
First thing I tried when investigating was turn off the Cloudflare proxy and clear the DNS cache. Surprisingly, the cross then worked. Next, I re-enabled the Cloudflare proxy and cleared the DNS cache again. The cross still worked. When I tried again a few minutes later, it was broken again...

So it seems Cloudflare proxy is breaking this? Any suggestions for further investigation?

**Comment by Immortality-IMT** [comment]
I have raised this issue previously, you have to copy and paste the query without using the keyboard to replicate the issue. As soon as you press the keyboard the cross appears.

**Comment by return42** [comment]
@Immortality-IMT [Your experience comes from an outdated copy of SearXNG that you are tinkering with](https://github.com/searxng/searxng/issues/3726#issuecomment-2282647078)

> Yes, it works fine on https://darmarit.org/searx/search?q=foo in Safari on Mac ..
> I experienced the issue with my instance .. The cross does not work there either, so it's probably not the browser, but something with (the settings of?) my instance.

@pro-sumer: Can you check the JavaScript files are loaded / I expect a `searxng.head.min.js` and a `searxng.min.js` to be loaded (use SHIFT-STRG-I to open debugger in your WEB browser):

![image](https://github.com/user-attachments/assets/b91b9be0-2404-473e-8876-691cf56f5a91)

**Comment by return42** [comment]
> I have raised this issue previously, you have to copy and paste the query without using the keyboard to replicate the issue.

The issue you report here is different to the issue the opener reports .. please refer to your ticket where reported your issue previously .. than we can re-open this ticket from yours.

**Comment by pro-sumer** [comment]
> Can you check the JavaScript files are loaded / I expect a searxng.head.min.js and a searxng.min.js to be loaded (use SHIFT-STRG-I to open debugger in your WEB browser)

I think so:

![JavaScript@2x](https://github.com/user-attachments/assets/93b48a29-db29-4493-9887-55b70ddc84cc)

PS: For those wondering; "STRG" is (German for?) what is known as "CTRL" in most parts of the world. However, for Safari on Mac the shortcut to show the Web Inspector is `⌥⌘I`.

**Comment by pro-sumer** [comment]
Does the difference in error message on the result page lead us anywhere?

**My private instance**
![CSP-private@2x](https://github.com/user-attachments/assets/c1ee7570-f7a9-434d-99b3-7536a5d8e3d9)

**Random public instance**
![CSP-public@2x](https://github.com/user-attachments/assets/f4dd3acd-6537-45bd-a295-74378dc35115)
